src/cpp/user_interface/arguments_configuration.cpp

```diff
@@ -0,0 +1,291 @@
+                    break;
+
+                case optionIndex::ACTIVE_DEBUG:
+                    active_debug = true;
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+    else
+    {
+        option::printUsage(fwrite, stdout, usage, columns);
+        return ProcessReturnCode::INCORRECT_ARGUMENT;
+    }
+
+    return ProcessReturnCode::SUCCESS;
+}
+
+void Arg::print_error(
+        const char* msg1,
+        const option::Option& opt,
+        const char* msg2)
+{
+    fprintf(stderr, "%s", msg1);
+    fwrite(opt.name, opt.namelen, 1, stderr);
+    fprintf(stderr, "%s", msg2);
+}
+
+option::ArgStatus Arg::Unknown(
+        const option::Option& option,
+        bool msg)
+{
+    if (msg)
+    {
+        print_error("Unknown option '", option, "'\nUse -h to see this executable possible arguments.\n");
+    }
+    return option::ARG_ILLEGAL;
+}
+
+option::ArgStatus Arg::Required(
+        const option::Option& option,
+        bool msg)
+{
+    if (option.arg != 0 && option.arg[0] != 0)
+    {
+        return option::ARG_OK;
+    }
+
+    if (msg)
+    {
+        print_error("Option '", option, "' requires an argument\n");
+    }
+    return option::ARG_ILLEGAL;
+}
+
+option::ArgStatus Arg::Numeric(
+        const option::Option& option,
+        bool msg)
+{
+    char* endptr = 0;
+    if (option.arg != 0 && std::strtol(option.arg, &endptr, 10))
+    {
+    }
+    if (endptr != option.arg && *endptr == 0)
+    {
+        return option::ARG_OK;
+    }
+
+    if (msg)
+    {
+        print_error("Option '", option, "' requires a numeric argument\n");
+    }
+    return option::ARG_ILLEGAL;
+}
+
+option::ArgStatus Arg::Float(
+        const option::Option& option,
+        bool msg)
+{
+    char* endptr = 0;
+    if (option.arg != 0 && std::strtof(option.arg, &endptr))
+    {
+    }
+    if (endptr != option.arg && *endptr == 0)
+    {
+        return option::ARG_OK;
+    }
+
+    if (msg)
+    {
+        print_error("Option '", option, "' requires a float argument\n");
+    }
+    return option::ARG_ILLEGAL;
+}
+
+option::ArgStatus Arg::String(
+        const option::Option& option,
+        bool msg)
+{
+    if (option.arg != 0)
+    {
+        return option::ARG_OK;
+    }
+    if (msg)
+    {
+        print_error("Option '", option, "' requires a text argument\n");
+    }
+    return option::ARG_ILLEGAL;
+}
+
+option::ArgStatus Arg::Readable_File(
+        const option::Option& option,
+        bool msg)
+{
+    if (option.arg != 0)
+    {
+// Windows has not unistd library, so to check if file is redeable use a different method
+#if defined(_WIN32)
+        if (_access( option.arg, 04 /*read only*/ )) != -1 )
+#else
+        if (access( option.arg, R_OK ) != -1)
+#endif // if defined(_WIN32)
+
+        {
+            return option::ARG_OK;
+        }
+    }
+    if (msg)
+    {
+        print_error("Option '", option, "' requires an existing file as argument\n");
```

Comment:
Print a more general message, as there may be other causes for the error.

Reply:
Repeated in this PR older versions